### ISSUE TITLE
Cantata misses cover art tag in the track display

### ISSUE THREAD

**Issue by mwigzell** [CDrummond/cantata#1677]
**Describe the bug**
When browsing my music collection of FLAC files using VORBIS tag for the art, a large percent of tracks with images are missed.The tracks all have an image tag embedded in each of them. I used "kid3" to tag the files. I have checked some having missing entries with the "easytag" tool and the images are defined in the file.

**To Reproduce**
Browse a Music folder using "cantata" 

**Expected behavior**
The flac files with images defined show that image

**Screenshots**
Often the image is missing

**Version Info**
 - OS: [linux]
 - Cantata Version: [2.4.2-1]
 - MPD Version: [0.22.4-2]
 - CMake options (other than default):

**Additional context**
Add any other context about the problem here. Art images are JPEG. Usually not very large (500x500 max)

**Comment by CDrummond** [comment]
Cantata does read some embedded images, but needs TagLib 1.11 for this. Which version is on your system?

**Comment by mwigzell** [comment]
Hi Craig, thanks for quick reply. I think this is it:
/usr/lib/libtag.so.1.17.0
so, not same version, but a newer one. I was reading into issue, it seems
"mpd" doesn't support the cover art tag, I hadn't realized. So it means you
are doing some extra work to read the "image" tag in the flac file? That's
great. I wonder why it isn't working.
I realize now, that the files I'm seeing that do have cover art, probably
have fetched the image from an internet source, correct?

**Comment by CDrummond** [comment]
Ho cantata gets covers depends on your setup. Can cantata acess the files directly? As in can the machine Cantata is on see the files, or is MPD on a remote system (e.g. a Pi)?

**Comment by mwigzell** [comment]
Its all on same machine. Cantata could see the files directly if it knew
where they were. Its all within same account and home directory too.

**Comment by CDrummond** [comment]
"Cantata **could** see the files directly **if** it knew where they were" ??? Not sure you mean by "could ... if" - have you configured Cantata with the correct MPD folder?

Is it Cantata reading the FLAC picture, or MPD?

**Comment by CDrummond** [comment]
`cantata --debug=covers` will cause cantata to log where it is looking for covers.

**Comment by mwigzell** [comment]
I meant, I didn't configure the location of my music files. I think it auto
detected mpd protocol too. Of course mpd knows where my music directory is.
I noticed that there is a parameter in ~/.config/cantata/cantata.cnf:
dir=/var/lib/mpd/music/
but that is not a valid dir. under /var/lib/mpd there is just a folder
`playlists'. But it is empty.
So not sure what good the setting is doing for Cantata.
Let me know what I should point it to. Perhaps Arch MPD is using an
alternate location for the mpd music files?

**Comment by mwigzell** [comment]
Ok, but I don't keep the covers separate or anything. They are all embedded
in the actual flac music files that they correspond to. Wasteful, but
simpler I thought.
What is Cantata expecting of the "covers" folder? What should it find there?

**Comment by mwigzell** [comment]
ah, I see! Got it!
I pointed the cantata.conf "dirs" entry to location of my music, then
Cantata was able to pull the images from the ".flac" file.
Thanks a bunch!
(also apologies, I only just discovered the "Preferences" UI, I kind of
jumped the gun a little in terms of reporting an issue.)
Btw. I really like Cantata esp. the shuffle feature. I need a reliable
replacement for Banshee, which is no longer maintained. I really like the
Information lookup on each album, its great!

One ask which I have, is: would it be possible to present just the cover
art with perhaps some info when the mouse hovers on one? The manner of
presentation of the art could be done in a straight forward 2D fashion like
Banshee and would be a total equalizer I think.
However it would be possible to cook up some real cool graphic presentation
of the covers, using a "shrinking/growing" kind of effect for instance,
perhaps presenting albums related by artist or genre etc. or random.

The reason I would like to peruse my collection like this is that it is
close to scanning through the collection by flipping, like in a record
store. Often times it was the cover art that inspired me to buy the album,
and I recognize the music by the cover most of the time.

Its a way to "grasp" the music which goes beyond looking at titles. I have
about 18000 tracks, so I have lots of music I don't play because it never
jumps out at me. I know, Cantata is in "maintenance" mode right now. But
I'd be interested in your take on such an approach to browsing a music
collection. Am I the only one that wants something like that? Banshee came
the closest.
Cheers!
--Mark
